packages/webpack-config-flag-plugin/test/fixtures/.eslintrc.js

```diff
@@ -6,6 +6,7 @@ module.exports = {
 		// Test fixtures import fake modules a lot just for testing the behaviour of this plugin
 		'import/no-extraneous-dependencies': 'off',
 		'wpcalypso/import-docblock': 0,
+		'wpcalypso/no-package-relative-imports': 'off',
```

Comment:
Makes sense. I think we can do:

- Use `no-package-relative-imports` in the root `.eslintrc.js` to make sure packages don't do `import 'lib/url'` by accident. It will force it to be `import 'calypso/lib/url'`. 
- Add an override for `packages/*` with the rules `no-restricted-modules` and `no-restricted-imports` to forbid imports from `calypso`.

The net effect of both rules is that doing `import 'config'` in a package will cause an eslint error. If a package really wants to do `import 'config'` (expecting `node_modules/config`), they will have to add an eslint exception for that line.

Once we have fixed all usages of `no-package-relative-imports` in master and all active branches then we can change webpack to make sure that import style is not even supported. At that point we can remove the exceptions to `no-package-relative-imports` (we can even disable/delete the rule, we won't need it anymore)

Note: `apps/notifications` also import from `calypso/...`

Reply:
> Note: `apps/notifications` also import from `calypso/...`

That's true and it's a shame 🙂 Fortunately these files don't really belong there and can be moved to a better location. Addressing in #47168.

Reply:
The eventual ideal state would be:
- packages use `import/no-extraneous-dependencies` to allow imports only from `package.json`-declared packages. That automatically flags imports like `calypso/lib/url` or `lib/url`, because neither `calypso` nor `lib` are declared dependencies. Makes `no-package-relative-imports` unneeded.
- the app in client supports only `calypso/*` imports, the unprefixed imports don't even work because the webpack alias has been removed. Then we only need to tell `import/no-extraneous-dependencies` that `calypso` is a valid pseudo-package. And again, `no-package-relative-imports` is no longer needed.

Does that sound right? Namely, that `no-package-relative-imports` rule is a temporary helper that assists the migration and will no longer be needed once that migration is fully done?

> The net effect of both rules is that doing `import 'config'` in a package will cause an eslint error.

That's an unfortunate effect and it would be nice if it was only temporary. If `config` is a valid dependency declared in `package.json` (the `test/e2e` uses `config` in exactly that meaning), it shouldn't cause an eslint error.

Reply:
We'll still need `no-restricted-modules/no-restricted-imports` to ensure packages don't add a dependency on `calypso`.

But other than that, yes once the migration is fully done (including changes in webpack so unprefixed imports don't even work) then the rule `no-package-relative-imports` can be disabled/deleted